File: Java/Find_The_Odd_Int.java
// Modified: 08/10/2017
//
// Author: jguardado39 (John Guardado)

/*



Given an array, find the int that appears an odd number of times.

There will always be only one integer that appears an odd number of times.

*/

public class FindOdd {
	public static int findIt(int[] A) {
    int i;
    int res = 0;
    for (i = 0; i < A.length; i++) {
      res = res ^ A[i];
    }
    return res;
  }
}

File: JavaScript/Find_The_Odd_Int.js
// Modified: 08/09/2017
//
// Author: jguardado39 (John Guardado)

/*
https://www.codewars.com/kata/54da5a58ea159efa38000836

Given an array, find the int that appears an odd number of times.

There will always be only one integer that appears an odd number of times.

*/

function findOdd(A) {
  var count = 0;
  for (var i = 0; i < A.length; i++) {
     count = count ^ A[i];
  }
  return count;
}

// cgb14d
function findOdd_alt(A) {
  return A.reduce(function(c,v){return c^v;},0);
}